frontend/src/components/Searchbar.js

```diff
@@ -0,0 +1,173 @@
+        filter_choice: this.reduxState.filter_choice,
+        isChecked: this.reduxState.isChecked
+      }
+    }
+
+    // Searchbar will either have a style or a size
+    this.SEARCHBAR_STYLE = this.props.style;
+    this.SEARCHBAR_SIZE = this.props.size;
+
+    this.autocomplete = null;
+
+    this.handleScriptLoad = this.handleScriptLoad.bind(this);
+    this.handlePlaceSelect = this.handlePlaceSelect.bind(this);
+    this.updateReduxState = this.updateReduxState.bind(this);
+    this.updateReduxStateValue = this.updateReduxStateValue.bind(this);
+  }
+
+  handleScriptLoad() {
+    const searchbar = document.getElementById(AUTOCOMPLETE_ID);
+    this.autocomplete = new google.maps.places.Autocomplete(searchbar, TYPES);
+    this.autocomplete.setFields(FIELDS);
+    this.autocomplete.addListener(PLACE_CHANGED_LISTENER, this.handlePlaceSelect);
+  }
+
+  async handlePlaceSelect() {
+    const addressObject = this.autocomplete.getPlace();
+    const address = addressObject.address_components;
+    const addressGeometry = addressObject.geometry;
+    const validPlusCode = typeof addressObject.plus_code != 'undefined';
+
+    if (address && validPlusCode) {
+      this.updateReduxState(addressObject, addressGeometry, addressObject.plus_code.global_code);
+      if (this.props.rerenderParentCallback) {
+        this.props.rerenderParentCallback();
+      }
+    } else {
+      ReactDOM.render(
+        <div>
+          <PopUp 
+            show={true}
+            onHide={this.hidePopUp.bind(this)}
+            message={INVALID_LOCATION_ALERT}/>
+        </div>
+      );
+      document.getElementById('popup-wrapper');
+    }
+  }
+
+  hidePopUp() {
+    const modal = document.getElementById('popup-wrapper');
+    ReactDOM.unmountComponentAtNode(modal);
+  }
+
+  async updateReduxState(addressObject, addressGeometry, plus_code_global_code) {
+    this.updateReduxStateValue();
+    this.setState(this.reduxState);
+    const updatedState = {
+      location: addressGeometry.location,
+      lat: addressGeometry.location.lat(),
+      lng: addressGeometry.location.lng(),
+      locationObject: addressObject,
+      plusCode: plus_code_global_code,
+      loggedIn: this.state.loggedIn,
+      credentials: this.state.credentials,
+      filter_choice: this.state.filter_choice,
+      isChecked: this.state.isChecked
+    };
+    await this.props.changeMapState(updatedState);
+    this.updateReduxStateValue();
+    this.setState(updatedState);
+  }
+
+  updateReduxStateValue() {
+    this.reduxState = this.props.articles[0];
+  }
+
+  render() {
+    if (this.SEARCHBAR_STYLE) {
+      return (
+        <div>
+          <Script
+            url={url}
+            onLoad = {this.handleScriptLoad}/>
+            <Form>
+              <Form.Control
+                style={this.SEARCHBAR_STYLE}
+                id={AUTOCOMPLETE_ID}
+                placeholder={SELECT_PLACEHOLDER}/>
+            </Form>
+        </div>
```

Comment:
Indentation here and the one below this is off

Reply:
I'm not sure why - it looks fine in my editor.  Is the indentation off by 1 or 2, or is it haphazard on your end?

Reply:
Yeah honestly I think github just screws with the indentation a bit and it would look different on different editors. You can disregard this change if it looks fine on your end.